mmaction/models/backbones/uniformer.py

```diff
@@ -0,0 +1,663 @@
+        self.patch_embed2 = PatchEmbed(
+            img_size=img_size // 4,
+            patch_size=2,
+            in_chans=embed_dim[0],
+            embed_dim=embed_dim[1])
+        self.patch_embed3 = PatchEmbed(
+            img_size=img_size // 8,
+            patch_size=2,
+            in_chans=embed_dim[1],
+            embed_dim=embed_dim[2])
+        self.patch_embed4 = PatchEmbed(
+            img_size=img_size // 16,
+            patch_size=2,
+            in_chans=embed_dim[2],
+            embed_dim=embed_dim[3])
+
+        self.pos_drop = nn.Dropout(p=drop_rate)
+        dpr = [
+            x.item() for x in torch.linspace(0, drop_path_rate, sum(depth))
+        ]  # stochastic depth decay rule
+        num_heads = [dim // head_dim for dim in embed_dim]
+        self.blocks1 = ModuleList([
+            CBlock(
+                dim=embed_dim[0],
+                mlp_ratio=mlp_ratio,
+                drop=drop_rate,
+                drop_path=dpr[i]) for i in range(depth[0])
+        ])
+        self.blocks2 = ModuleList([
+            CBlock(
+                dim=embed_dim[1],
+                mlp_ratio=mlp_ratio,
+                drop=drop_rate,
+                drop_path=dpr[i + depth[0]]) for i in range(depth[1])
+        ])
+        self.blocks3 = ModuleList([
+            SABlock(
+                dim=embed_dim[2],
+                num_heads=num_heads[2],
+                mlp_ratio=mlp_ratio,
+                qkv_bias=qkv_bias,
+                qk_scale=qk_scale,
+                drop=drop_rate,
+                attn_drop=attn_drop_rate,
+                drop_path=dpr[i + depth[0] + depth[1]],
+                norm_layer=norm_layer) for i in range(depth[2])
+        ])
+        self.blocks4 = ModuleList([
+            SABlock(
+                dim=embed_dim[3],
+                num_heads=num_heads[3],
+                mlp_ratio=mlp_ratio,
+                qkv_bias=qkv_bias,
+                qk_scale=qk_scale,
+                drop=drop_rate,
+                attn_drop=attn_drop_rate,
+                drop_path=dpr[i + depth[0] + depth[1] + depth[2]],
+                norm_layer=norm_layer) for i in range(depth[3])
+        ])
+        self.norm = bn_3d(embed_dim[-1])
+
+        self.apply(self._init_weights)
+
+        self._load_pretrained(pretrained)
+
+    def _inflate_weight(self,
+                        weight_2d: torch.Tensor,
+                        time_dim: int,
+                        center: bool = True) -> torch.Tensor:
+        logger.info(f'Init center: {center}')
+        if center:
+            weight_3d = torch.zeros(*weight_2d.shape)
+            weight_3d = weight_3d.unsqueeze(2).repeat(1, 1, time_dim, 1, 1)
+            middle_idx = time_dim // 2
+            weight_3d[:, :, middle_idx, :, :] = weight_2d
+        else:
+            weight_3d = weight_2d.unsqueeze(2).repeat(1, 1, time_dim, 1, 1)
+            weight_3d = weight_3d / time_dim
+        return weight_3d
+
+    def _init_weights(self, m):
```

Comment:
As a convention of OpenMMLab repo, we use `init_cfg` to specify weight initialization. For complicated situation `init_cfg` can't cover，we override `init_weights` and automatically call it in `Runner`.

Reply:
Done